docker-compose.yml

```diff
@@ -0,0 +1,174 @@
+    ]
+    expose:
+      - "2181"
+    environment:
+      LOG_DIR: /tmp/logs
+    volumes:
+      - ./compose-data/zookeeper:/tmp/logs
+  
+  kafka:
+    profiles:
+     - "kafka"
+    image: quay.io/strimzi/kafka:latest-kafka-3.4.0
+    command: [
+      "sh", "-c",
+      "bin/kafka-server-start.sh config/server.properties  --override inter.broker.listener.name=$${KAFKA_INTER_BROKER_LISTENER_NAME} --override listener.security.protocol.map=$${KAFKA_LISTENER_SECURITY_PROTOCOL_MAP} --override listeners=$${KAFKA_LISTENERS} --override advertised.listeners=$${KAFKA_ADVERTISED_LISTENERS} --override zookeeper.connect=$${KAFKA_ZOOKEEPER_CONNECT}"
+    ]
+    expose:
+      - 9092
+      - 29092
+    depends_on:
+      - zookeeper
+    environment:
+      KAFKA_AUTO_CREATE_TOPICS_ENABLE: true
+      LOG_DIR: "/tmp/logs"
+      KAFKA_BROKER_ID: 1
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      KAFKA_INTER_BROKER_LISTENER_NAME: DOCKER
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: DOCKER:PLAINTEXT,LOCALHOST:PLAINTEXT
+      KAFKA_LISTENERS: DOCKER://kafka:29092,LOCALHOST://localhost:9092
+      KAFKA_ADVERTISED_LISTENERS: DOCKER://kafka:29092,LOCALHOST://localhost:9092
+    volumes:
+      - ./compose-data/kafka:/tmp/logs
+
+  init-kafka:
+    profiles:
+      - "kafka"
+    image: quay.io/strimzi/kafka:latest-kafka-3.4.0
+    entrypoint: [ '/bin/sh', '-c' ]
+    command: |
+      "
+      # blocks until kafka is reachable
+      bin/kafka-topics.sh --bootstrap-server kafka:29092 --list
+
+      echo -e 'Creating kafka topics'
+      bin/kafka-topics.sh --bootstrap-server kafka:29092 --create --if-not-exists --topic platform.sources.status --replication-factor 1 --partitions 1
+      bin/kafka-topics.sh --bootstrap-server kafka:29092 --create --if-not-exists --topic platform.provisioning.internal.availability-check --replication-factor 1 --partitions 1
+
+      echo -e 'Successfully created the following topics:'
+      bin/kafka-topics.sh --bootstrap-server kafka:29092 --list
+      "
+    depends_on:
+      - kafka
+      - zookeeper
+  
+  backend:
+    env_file:
+      - scripts/compose/api.compose.env
+    depends_on:
+      migrate:
+        condition: service_completed_successfully
+      provisioning-db:
+        condition: service_healthy
+      init-kafka:
+        condition: service_completed_successfully
+    build:
+      context: .
+      dockerfile: build/Dockerfile.dev
+    ports:
+      - ${APP_PORT:-8000}:${APP_PORT:-8000}
+      - ${PROMETHEUS_PORT:-9000}:${PROMETHEUS_PORT:-9000}
+    volumes:
+      - ./:/backend
+
+  migrate: # This service is used to run migrations
+    profiles:
+      - "migrate"
+    build:
+      context: .
+      dockerfile: build/Dockerfile.dev
+    env_file:
+      - config/api.env
+    depends_on:
+      provisioning-db:
+        condition: service_healthy
+    entrypoint: [go,run,/build/cmd/pbackend migrate]
```

Comment:
Kafka is named "init-kafka", maybe this could be also named "init-backend". Similarily, kafka initialization is in the "kafka" profile, I think this could also be in the "backend" profile too, there is no need to have this separately migration is very quick.

Docker and env file paths needs to be updated, see above.